### PR TITLE
S05 us154 character entity

### PR DESCRIPTION
**Pull request by boubseb**
attendre les autres pr

### CODE REVIEW

File: src/main/java/com/example/demo/controller/charactersController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        if (characterOptional.isPresent()) {
+            Character character = characterOptional.get();
+            characterRepository.delete(character);
+            return ResponseEntity.ok(character);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    // @GetMapping("user/{id}")
```
Comment (OCR-VH-DevWeb) [inline src/main/java/com/example/demo/controller/charactersController.java]: Dead code ? in this case delete or add a comments to explain why you keep this comments

File: src/main/java/com/example/demo/entity/Character.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+   @ManyToMany
+   @JoinTable(
+        name = "character_title",
+        joinColumns = @JoinColumn(name = "characterId"),
+        inverseJoinColumns = @JoinColumn(name = "titleId")
+    )
+   private Set<Title> titles;
+
+
+   @ManyToOne
```
Comment (OCR-VH-DevWeb) [inline src/main/java/com/example/demo/entity/Character.java:51]: Miss many of attributs for the character sheet game to add later. But for the presentation it's ok

**Review by OCR-VH-DevWeb — APPROVED** [review]

**Review by OCR-VH-DevWeb — APPROVED** [review]

File: src/main/java/com/example/demo/controller/charactersController.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            Character character = characterOptional.get();
+        
+            return ResponseEntity.ok(character);
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Character> deleteCharacter(@PathVariable Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/charactersController.java]: token required to delete character ?

File: src/main/java/com/example/demo/controller/charactersController.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        this.characterRepository = characterRepository;
+    }
+
+
+    @PutMapping("/addCharacter")
+    public void addCharacter(@RequestHeader("Authorization") String BearerHeader) throws StreamReadException, DatabindException, IOException {     
+    }
+
+        @GetMapping("/{id}")
+    public ResponseEntity<Character> getCharacterById(@PathVariable Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/charactersController.java]: does anyone can consult characters ?

File: src/main/java/com/example/demo/entity/BeliefContent.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    
+
+
+    public BeliefContent(Integer number, String content, Kingdom kingdom) {
+        this.number = number;
+        this.content = content;
+        this.kingdom = kingdom;
+    }
+
+
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/BeliefContent.java]: no getter for id ?

File: src/main/java/com/example/demo/controller/charactersController.java
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/charactersController.java:1]: use service methods already defined rather than repository here

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/controller/charactersController.java (hunk 4 above)
Comment (boubseb) [reply src/main/java/com/example/demo/controller/charactersController.java]: I think yes if the profil is public and we will have to use DTO to not display secrets

File: src/main/java/com/example/demo/controller/charactersController.java [hunk 6]
```diff
@@ -26,47 +26,43 @@
...
-        this.characterRepository = characterRepository;
-    }
+        @Autowired
+        private CharacterService characterService;
+        private UserService userService;
+
+  
 
 
     @PutMapping("/addCharacter")
```
Comment (Benja-san) [inline src/main/java/com/example/demo/controller/charactersController.java]: this is a post mapping right ? where's the update ?

**Review by Benja-san — APPROVED** [review]

**Pull request merged**